### PR TITLE
Move to CMake

### PR DESCRIPTION
**Pull request by cuzbog**
@iBeliever please test this as I don't have the required libs on my system, cmake builds fine though

### CODE REVIEW

File: src/cpp/qmlcompositor.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ QmlCompositor::QmlCompositor()
     : QWaylandQuickCompositor("papyros-shell", DefaultExtensions | SubSurfaceExtension)
     , m_fullscreenSurface(0)
 {
-    setSource(QUrl("qrc:/qml/main.qml"));
+    setSource(QUrl("/usr/share/papyros-shell/qml/main.qml"));
```
Comment (iBeliever) [inline src/cpp/qmlcompositor.cpp]: This won't work if the user installs the shell to `/opt`, for example.

**Comment by boghison** [comment]
@iBeliever That would mean a custom install (cmakelists + cpp), I don't think qmake provides custom locations either, but the advantage is that the qml files are left intact and the user can configure them :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    ${CMAKE_SOURCE_DIR}/src/cpp/main.cpp
+    ${CMAKE_SOURCE_DIR}/src/cpp/qmlcompositor.cpp
+    ${CMAKE_SOURCE_DIR}/src/cpp/qmlcompositor.h
+)
+
+ADD_EXECUTABLE(papyros-shell ${SHELL_SRCS})
+TARGET_LINK_LIBRARIES(papyros-shell Qt5::Quick)
+TARGET_LINK_LIBRARIES(papyros-shell Qt5::Compositor)
+
+INSTALL(TARGETS papyros-shell DESTINATION /usr/bin)
```
Comment (iBeliever) [inline CMakeLists.txt]: These paths shouldn't be hard-coded to `/usr`.

**Comment by boghison** [comment]
@iBeliever You mean the executable install path?

**Comment by iBelieve** [comment]
The project is now CMake based, using the Green Island installation step.

**Pull request closed without merging**